### PR TITLE
feat: add a custom class-validator decorator to make custom validation

### PR DESCRIPTION
**Pull request by maxmousse**

### CODE REVIEW

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { CustomValidate } from './custom-validate';
+
+// Create a mock predicate
+const mockedPredicate = jest.fn();
+
+// Class to test the CustomValidate decorator
+class Car {
+  @CustomValidate([mockedPredicate])
+  brand!: string;
+}
```
Comment (floross) [inline libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: That should be inside the describe function

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+// Create an instance of Car
+const car = new Car();
+car.brand = 'Ford';
```
Comment (floross) [inline libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: That should be inside the before each

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  it('should call the predicate function passed in parameter with the object being validated', async () => {
+    mockedPredicate.mockReturnValue(true);
+
+    await validate(car);
+
+    expect(mockedPredicate).toHaveBeenCalledWith(car);
+  });
+
+  it('should throw an error with a message if the predicate function passed in parameter return false', async () => {
```
Comment (floross) [inline libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: Should throw ? no throw has been catch inside this test

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    mockedPredicate.mockReturnValue('NotABoolean');
+
+    await expect(async () => validate(car)).rejects.toThrow(
+      'Invalid constraint',
+    );
+  });
+
+  it('should throw an error if no predicates is passed to the decorator', async () => {
+    class Car2 {
+      @CustomValidate([])
```
Comment (floross) [inline libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: Maybe we should fail fast, inside the validator function and the constraint to

**Review by floross — CHANGES_REQUESTED** [review]

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts (hunk 1 above)
Comment (maxmousse) [reply libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: Done

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts (hunk 2 above)
Comment (maxmousse) [reply libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: Done

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts (hunk 3 above)
Comment (maxmousse) [reply libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: Done

File: libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts (hunk 4 above)
Comment (maxmousse) [reply libs/common/src/lib/validator-transformer/validators/custom-validate.spec.ts]: Done

**Review by floross — APPROVED** [review]

**Pull request merged**